mosip-identity-plugin/src/main/java/io/mosip/signup/plugin/mosipid/service/IdrepoProfileRegistryPluginImpl.java

```diff
@@ -160,15 +168,186 @@ public void init() {
+
+            // Skip if required fields are missing or documentTypes is not an array
+            if (categoryCode == null || langCode == null || documentTypes == null || !documentTypes.isArray())
+                continue;
+
+            // Get or create the node for the document category code
+            ObjectNode docTypeMap = (ObjectNode) result.get(categoryCode);
+            if (docTypeMap == null) {
+                docTypeMap = objectMapper.createObjectNode();
+                result.set(categoryCode, docTypeMap);
+            }
+
+            // Iterate through each document type and add to the nested structure
+            for (JsonNode docType : documentTypes) {
+                String docTypeCode = docType.hasNonNull("code") ? docType.get("code").asText() : null;
+                String docTypeName = docType.hasNonNull("name") ? docType.get("name").asText() : null;
+                if (docTypeCode == null || docTypeName == null) continue;
+
+                ObjectNode langMap = (ObjectNode) docTypeMap.get(docTypeCode);
+                if (langMap == null) {
+                    langMap = objectMapper.createObjectNode();
+                    docTypeMap.set(docTypeCode, langMap);
+                }
+                langMap.put(langCode, docTypeName);
+            }
+        }
+    }
+
+    public String buildDynamicFieldsUrl(int pageNumber, int pageSize) {
+        return String.format("%s?pageNumber=%d&pageSize=%d", dynamicFieldsBaseUrl, pageNumber, pageSize);
+    }
+    public String buildDocumentTypeAndCategoryUrl(List<String> languages) {
+        StringBuilder urlBuilder = new StringBuilder(docTypesAndCategoryBaseUrl);
+        urlBuilder.append("?");
+        for (int i = 0; i < languages.size(); i++) {
+            if (i != 0) urlBuilder.append("&");
+            urlBuilder.append("languages=").append(URLEncoder.encode(languages.get(i), StandardCharsets.UTF_8));
+        }
+        return urlBuilder.toString();
+    }
+
+    public List<String> getAllConfiguredLanguages() {
+        Set<String> allLanguages = new LinkedHashSet<>();
+        if (mandatoryLanguages != null) {
+            allLanguages.addAll(mandatoryLanguages);
+        }
+        if (optionalLanguages != null) {
+            allLanguages.addAll(optionalLanguages);
+        }
+        return new ArrayList<>(allLanguages);
+    }
+
+    /**
+     * fetch Document Types and categories from master data
+     * @param languages List<String> languages
+     * @return List of JSON nodes representing document categories and types
+     */
+    public List<JsonNode> fetchDocTypesAndCategories(List<String> languages) {
+        List<JsonNode> allFields = new ArrayList<>();
+        String url = this.buildDocumentTypeAndCategoryUrl(languages);
+        ResponseEntity<JsonNode> response = this.restTemplate.getForEntity(url, JsonNode.class);
+        JsonNode data = Objects.requireNonNull(response.getBody()).get("response").get("documentCategories");
```

Comment:
Potential NPE if the "Objects.requireNonNull(response.getBody()).get("response")" returns null @Md-Humair-KK